Tidy up the Readme.md; add quick start section (#14)
digital-asset/daml-platform-observability-example: README.md

# [Daml Enterprise](https://www.digitalasset.com/products/daml-enterprise) - Observability Example

This project demonstrates a [Daml
Enterprise](https://www.digitalasset.com/products/daml-enterprise)
deployment's observability features, along with example Grafana dashboards.  

> The project is self contained:  providing load generations scripts to generate

Reviewer: I would not call the scripts we have here `load generation scripts`, as it could imply that the scripts can be used to "load test" canton, while the scripts we have just generate demo data.

Author: Changed the phrase "load generation scripts."

# [Daml Enterprise](https://www.digitalasset.com/products/daml-enterprise) - Observability Example

This project demonstrates a [Daml
Enterprise](https://www.digitalasset.com/products/daml-enterprise)
deployment's observability features, along with example Grafana dashboards.  

> The project is self contained:  providing scripts to generate requests for
> which metrics are collected for display in the Grafana  dashboards.  

Reviewer 2: Nit:
```suggestion
The project is self-contained: providing scripts to generate requests for
which metrics are collected for display in the Grafana dashboards.  
```

> This project is provided for illustrative purposes and local use only. The

Reviewer 2: Please keep the disclaimer here that this is not production quality and/or may not fit any use-case (I'd rather avoid the expression 'local use only' as it may be ambiguous local to what):
```suggestion
*Please bear in mind that this project is provided for illustrative purposes only,
and as such may not be production quality and/or may not fit your use-cases.  The
```

samples have been tested on MacOS and Linux -- some Docker images may not be

> available for Windows. This repository does not accept Pull Requests at the

Reviewer 2: I'd suggest separating the statement on pull requests so that it is more obvious (i.e., to the last sentence in this section). You may want to emphasize it like [here](https://github.com/digital-asset/dev-env#template).

Author: I accepted your suggestion. Good to go.

> OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH
> THE USE OR PERFORMANCE OF THIS SOFTWARE.

## TL;DR Quick Start

To quickly get up and running, make sure you have the [prerequisites](#Prerequisites) installed
and then:
1. Deploy the containers locally, which include several Daml Enterprise
components, Prometheus, Grafana, and related containers.  This is done withe
command:
```docker compose -p obs up```.

> 2. Start the load generation.  There are three scripts that generate the load

Reviewer: Same here, let's use `demo data generation` or something similar and stay clear from the `load` terms.

>    and running them in separate consoles is best.  The scripts are:

Reviewer: Using the term consoles can be confusing as it might make users think of running them in the canton console.

Author: Fixed now.

Author: changed to 'shell console'

> OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH
> THE USE OR PERFORMANCE OF THIS SOFTWARE.

> **This repository does not accept Pull Requests at the moment.**

## Components

You can find a Docker Compose setup here with the following services:

* Daml Platform:
  * All-in-one Canton node (domain topology manager, mediator, sequencer and participant)
  * HTTP JSON API service
  * PostgreSQL database (used by the Canton node and the HTTP JSON API service)
* Monitoring:
  * Prometheus `2.x`
  * Grafana `9.x`
  * Node Exporter
  * Loki + Promtail `2.x`

Prometheus and Loki are [preconfigured as datasources for Grafana](./grafana/datasources.yml). You can add other
services/exporters in the [Docker compose file](./docker-compose.yml) and scrape them changing the
[Prometheus configuration](./prometheus/prometheus.yml).

> Use the `Makefile` — run `make help` for available commands!

Reviewer 3: Not sure why this whole section got removed, if you want to implement your own monitoring, it's important to know which software is used in the monitoring stack and they interact with each other

Author: My bad.  Somehow I lost it.  I put it back.  Thank you for pointing this out.

Author: Actually, I see it there so I don't know why it doesn't appear for you.  It was embellished with more information though.

> * ```scripts/generate-grpc-load.sh <number loops>``` this runs the ledger API

Reviewer 2: Suggestion to visually separate the script from its description:
```suggestion
* ```scripts/generate-grpc-load.sh <number loops>```: this runs the ledger API
```

> * ```scripts/send-json-api-requests.sh``` this generates HTTP traffic against

Reviewer 2: Suggestion to visually separate the script from its description:
```suggestion
* ```scripts/send-json-api-requests.sh```: this generates HTTP traffic against
```

Author: done

>   the HTTP JSON API server which has successful and unsuccessful requests.  It

Reviewer 2: ```suggestion
  the [HTTP JSON API Service](https://docs.daml.com/json-api/) which has successful and unsuccessful requests.  It
```

>   will loop continuously until stopped with \<ctrl-c\>.

Reviewer 2: Consistency:
```suggestion
  will loop continuously until stopped with `[Ctrl]+[c]`.
```

> * ```scripts/send-trigger-requests.sh``` generates traffic for the trigger service.  It

Reviewer 2: Suggestion to visually separate the script from its description:
```suggestion
* ```scripts/send-trigger-requests.sh```: generates traffic for the [Trigger Service](https://docs.daml.com/tools/trigger-service/index.html).  It
```

>   will loop continuously until stopped with \<ctrl-c\>.

Reviewer 2: ```suggestion
  will loop continuously until stopped with `[Ctrl]+[c]`.
```

* Artifactory credentials to access our private
[Canton Enterprise Docker images](https://digitalasset.jfrog.io/ui/repos/tree/General/canton-enterprise-docker/digitalasset/canton-enterprise/latest)

Docker Compose will automagically build the [image for the HTTP JSON API service and Trigger service](daml-service/)
from the release JAR files.

Docker Compose will automagically build the [image for the HTTP JSON API service](daml-service/) from the release JAR file.
Check the [`.env`](./.env) file to know which Canton and SDK versions are being used.

⚠️ You can test different versions changing `CANTON_VERSION` and `SDK_VERSION`, however, there is no guarantee that it will be

> compatible with the currently committed configurations and Grafana dashboards.

Reviewer 3: This warning got lost somehow, we need to keep it

Author: My bad.  Somehow I lost it.  I put it back.  Thank you for pointing this out.

Reviewer 2: We should just refer to the section below ('Accessing Daml Enterprise Docker Images') as this is described well there.

Reviewer 2: This sentence regarding how to test different versions is actually now in the TLDR section, so it hasn't been lost.

* Node exporter full [[source]](https://grafana.com/grafana/dashboards/1860-node-exporter-full/)
* Loki and Promtail [[source]](https://grafana.com/grafana/dashboards/14055-loki-stack-monitoring-promtail-loki/)

## Accessing [Daml Enterprise](https://www.digitalasset.com/products/daml-enterprise) Docker Images

1. Get an access key from digital asset.  This step needs to be performeed
  once.  If you haven't logged in before, you can get an access key by logging
  in to [digitalasset.jfrog.io](https://digitalasset.jfrog.io) using your

>   Google account and generating an identity token on your [Artifactory profile

Reviewer 2: I'm not sure if having a Google account is a requirement, i.e., you may log in to `digitalasset.jfrog.io` using a simple username/password pair as well.

Author: Removed reference to Google account
